server/src/test/java/keywhiz/service/daos/AclDAOTest.java

```diff
@@ -90,6 +94,26 @@
     secret2 = secretFixtures.createSecret("secret2", "c2VjcmV0Mg==");
   }
 
+  @Test public void listingRenamedSecretDoesNotFailHmacCheck() {
+    aclDAO.enrollClient(jooqContext.configuration(), client1.getId(), group1.getId());
+    aclDAO.allowAccess(jooqContext.configuration(), secret1.getId(), group1.getId());
+
+    String newName = UUID.randomUUID().toString();
+    secretSeriesDAO.renameSecretSeriesById(
+        secret1.getId(),
+        newName,
+        client1.getName(),
+        ApiDate.now().toEpochSecond());
+
+    KeywhizConfig.RowHmacCheck originalHmacCheck = keywhizConfig.getRowHmacCheck();
+    try {
+      keywhizConfig.setRowHmacCheck(KeywhizConfig.RowHmacCheck.ENFORCED);
+      aclDAO.getSanitizedSecretsFor(client1);
+    } finally {
+      keywhizConfig.setRowHmacCheck(originalHmacCheck);
+    }
+  }
```

Comment:
This test initially failed before the bug was fixed. This is a higher-level unit test that mimics the actual calls that were in the initial bug report.